storage/remote/write.go

```diff
@@ -34,6 +34,7 @@ import (
 	"github.com/prometheus/prometheus/tsdb/wlog"
 )
 
+// TODO: needed?
```

Comment:
we already have those on queuemanager level (..._pending_...), removing them here would help get rid of the whole `run()` logic.